zilliqa/src/message.rs

```diff
@@ -439,6 +460,16 @@ impl Committee {
     pub fn add_validators(&mut self, validators: impl IntoIterator<Item = Validator>) {
         self.0.extend(validators);
     }
+
+    pub fn remove_by_peer_id(&mut self, peer_id: PeerId) {
+        self.0.retain(|v| v.peer_id != peer_id);
+    }
+
+    pub fn choose_random(&mut self) -> Validator {
```

Comment:
❗ - I really like that in the tests everything is deterministic. This potentially makes it not so - when you are requesting someone to update you with the chain, you request another peer with true random probability.

Is there a way to do this nicely with libp2p? Even so I suspect that it would be random, still. Alternatives are to request based on an xor of your peer id with chain height or something, seems hacky though.

Reply:
I think it's too convoluted atm. Maybe lets see what James thinks when get gets back. Now that I think of it, another way you could do it would be to override the random import, much like we do with systemtime. That way you can use random freely in the code.

Reply:
I agree that keeping the tests deterministic is important.

I would probably reach for adding an `Rng` to `Consensus` as the simplest solution, since doing the import-mocking trick requires quite a bit more fiddling with features, etc. and is less immediately understandable.

Reply:
^ will perform this in a follow up PR